src/components/Tooltip/Tooltip.tsx

```diff
@@ -0,0 +1,347 @@
+          case 'left':
+            positionLeft(tooltipBody)
+            if (!isElementInViewport(tooltipBody)) {
+              positionRight(tooltipBody)
+              if (!isElementInViewport(tooltipBody)) {
+                positionTop(tooltipBody)
+              }
+            }
+            break
+
+          default:
+            // skip default case
+            break
+        }
+      }
+    }, [isVisible])
+  }
+
+  if (isCustomProps(props)) {
+    const triggerElementRef = useRef<HTMLElement>(null)
+    const { label, position, asCustom, children, ...remainingProps } = props
+    const customProps: FCProps = (remainingProps as unknown) as FCProps
+
+    const tooltipClasses = classnames('usa-tooltip__body', {
+      'is-set': isVisible,
+      'usa-tooltip__body--top': position === 'top',
+      'usa-tooltip__body--bottom': position === 'bottom',
+      'usa-tooltip__body--right': position === 'right',
+      'usa-tooltip__body--left': position === 'left',
+      'is-visible': isVisible,
+    })
+    const activateTooltip = (): void => {
+      setVisible(true)
+    }
+    const deactivateTooltip = (): void => {
+      setVisible(false)
+    }
+
+    useTooltip(triggerElementRef, position)
+
+    const triggerElement = createElement(
+      asCustom,
+      {
+        ref: triggerElementRef,
+        'data-testid': 'triggerElement',
+        onMouseEnter: () => activateTooltip(),
+        onMouseOver: () => activateTooltip(),
+        onFocus: () => activateTooltip(),
+        onMouseLeave: () => deactivateTooltip(),
+        onBlur: () => deactivateTooltip(),
+        onKeyDown: () => deactivateTooltip(),
+        ...customProps,
+      },
+      children
+    )
+
+    return (
+      <span
+        data-testid="tooltipWrapper"
+        ref={wrapperRef}
+        className="usa-tooltip"
+        {...customProps}
+        role="tooltip">
+        {triggerElement}
+        <span
+          data-testid="tooltipBody"
+          title={label}
+          id={tooltipID}
+          ref={tooltipBodyRef}
+          className={tooltipClasses}>
+          {label}
+        </span>
+      </span>
+    )
+  } else {
+    const triggerElementRef = useRef<HTMLButtonElement>(null)
+    const { label, position, children, ...spanProps } = props
+    const tooltipClasses = classnames('usa-tooltip__body', {
+      'is-set': isVisible,
+      'usa-tooltip__body--top': position === 'top',
+      'usa-tooltip__body--bottom': position === 'bottom',
+      'usa-tooltip__body--right': position === 'right',
+      'usa-tooltip__body--left': position === 'left',
+      'is-visible': isVisible,
+    })
+
+    const activateTooltip = (): void => {
+      setVisible(true)
+    }
+    const deactivateTooltip = (): void => {
+      setVisible(false)
+    }
+
+    useTooltip(triggerElementRef, position)
+
+    return (
+      <span
+        data-testid="tooltipWrapper"
+        ref={wrapperRef}
+        className="usa-tooltip"
+        {...spanProps}
```

Comment:
I think the remaining props are meant to be passed onto the button element, rather than the wrapper span (this should mirror the custom implementation above)